Source/EasyNetQ/Producer/PersistentChannel.cs

```diff
@@ -36,136 +36,160 @@ public class PersistentChannel : IPersistentChannel
             WireUpEvents();
         }
 
-        private void WireUpEvents()
-        {
-            eventBus.Subscribe<ConnectionDisconnectedEvent>(OnConnectionDisconnected);
-            eventBus.Subscribe<ConnectionCreatedEvent>(ConnectionOnConnected);
-        }
-
-        private void OnConnectionDisconnected(ConnectionDisconnectedEvent @event)
-        {
-            if (!disconnected)
-            {
-                disconnected = true;
-                channel = null;
-                logger.DebugWrite("Persistent channel disconnected.");
-            }
-        }
-
-        private void ConnectionOnConnected(ConnectionCreatedEvent @event)
+        public void InvokeChannelAction(Action<IModel> channelAction)
         {
-            if (connection.IsConnected)
+            Preconditions.CheckNotNull(channelAction, "channelAction");
+            var startTime = DateTime.UtcNow;
+            var retryTimeout = TimeSpan.FromMilliseconds(50);
+            while (!IsTimedOut(startTime))
             {
                 try
                 {
-                    OpenChannel();
+                    var channel = OpenChannel();
+                    channelAction(channel);
+                    return;
+                }
+                catch (OperationInterruptedException exception)
+                {
+                    CloseChannel();
+                    if (IsStopError(exception))
```

Comment:
probably better to call it `IsBlockingException(exception)` or `ShouldRethrow(exception)`?